applications/StructuralMechanicsApplication/custom_constitutive/small_strain_isotropic_damage_factory_3d.h

```diff
@@ -0,0 +1,427 @@
+            if (potential == "VonMises") {
+                return GenericSmallStrainIsotropicDamage3D 
+                    <GenericConstitutiveLawIntegratorDamage
+                        <ModifiedMohrCoulombYieldSurface
+                            <VonMisesPlasticPotential>>>().Clone()
+                ;
+            } else if (potential == "ModifiedMohrCoulomb") {
+                return GenericSmallStrainIsotropicDamage3D 
+                    <GenericConstitutiveLawIntegratorDamage
+                        <ModifiedMohrCoulombYieldSurface
+                            <ModifiedMohrCoulombPlasticPotential>>>().Clone()
+                ;
+            } else if (potential == "DruckerPrager") {
+                return GenericSmallStrainIsotropicDamage3D 
+                    <GenericConstitutiveLawIntegratorDamage
+                        <ModifiedMohrCoulombYieldSurface
+                            <DruckerPragerPlasticPotential>>>().Clone()
+                ;
+            } else if (potential == "Tresca") {
+                return GenericSmallStrainIsotropicDamage3D 
+                    <GenericConstitutiveLawIntegratorDamage
+                        <ModifiedMohrCoulombYieldSurface
+                            <TrescaPlasticPotential>>>().Clone()
+                ;
+            } else {
+                KRATOS_ERROR << "Plastic Potential not defined or wrong" << std::endl;
+            }
+        } else if (yield == "Tresca") {
+            if (potential == "VonMises") {
+                return GenericSmallStrainIsotropicDamage3D 
+                    <GenericConstitutiveLawIntegratorDamage
+                        <TrescaYieldSurface
+                            <VonMisesPlasticPotential>>>().Clone()
+                ;
+            } else if (potential == "ModifiedMohrCoulomb") {
+                return GenericSmallStrainIsotropicDamage3D 
+                    <GenericConstitutiveLawIntegratorDamage
+                        <TrescaYieldSurface
+                            <ModifiedMohrCoulombPlasticPotential>>>().Clone()
+                ;
+            } else if (potential == "DruckerPrager") {
+                return GenericSmallStrainIsotropicDamage3D 
+                    <GenericConstitutiveLawIntegratorDamage
+                        <TrescaYieldSurface
+                            <DruckerPragerPlasticPotential>>>().Clone() 
+                ;
+            } else if (potential == "Tresca") {
+                return GenericSmallStrainIsotropicDamage3D 
+                    <GenericConstitutiveLawIntegratorDamage
+                        <TrescaYieldSurface
+                            <TrescaPlasticPotential>>>().Clone()
+                ;
+            } else {
+                KRATOS_ERROR << "Plastic Potential not defined or wrong" << std::endl;
+            }
+        } else if (yield == "DruckerPrager") {
+            if (potential == "VonMises") {
+                return GenericSmallStrainIsotropicDamage3D 
```

Comment:
as i told you orally @pooyan-dadvand suggests registering all of those

the advantage of doing so is that one can add new combinations outside of this file, or even in a derive application

Reply:
this is work in progress, I'll implement it as we told.